### ISSUE TITLE
SwiftLint v0.54.0 fails to be installed with Mint

### ISSUE THREAD

**Issue by asa08** [realm/SwiftLint#5348]
### New Issue Checklist

- [x] Updated SwiftLint to the latest version
- [x] I searched for [existing GitHub issues](https://github.com/realm/SwiftLint/issues)

### Describe the bug

An error occurs when executing the following commands.

```
mint bootstrap
```

##### Complete output when running SwiftLint, including the stack trace and command used

```bash
$ mint bootstrap
🌱 Cloning SwiftLint 0.54.0
🌱 Resolving package
🌱 Building product swiftlint
Building for production...
[0/11] Compiling SourceKit SourceKit.c
[1/11] Compiling Clang_C Clang_C.c
remark: Incremental compilation has been disabled: it is not compatible with whole module optimization
remark: Incremental compilation has been disabled: it is not compatible with whole module optimization
[3/13] Compiling SwiftSyntax509 Empty.swift
remark: Incremental compilation has been disabled: it is not compatible with whole module optimization
[3/13] Compiling DyldWarningWorkaround DyldWarningWorkaround.c
remark: Incremental compilation has been disabled: it is not compatible with whole module optimization
[4/15] Compiling writer.c
remark: Incremental compilation has been disabled: it is not compatible with whole module optimization
[5/16] Compiling parser.c
remark: Incremental compilation has been disabled: it is not compatible with whole module optimization
[6/16] Compiling reader.c
[7/17] Compiling emitter.c
[8/17] Compiling scanner.c
[9/17] Compiling api.c
remark: Incremental compilation has been disabled: it is not compatible with whole module optimization
[11/18] Compiling CollectionConcurrencyKit CollectionConcurrencyKit.swift
[12/18] Compiling ArgumentParserToolInfo ToolInfo.swift
remark: Incremental compilation has been disabled: it is not compatible with whole module optimization
[13/19] Compiling SwiftyTextTable TextTable.swift
[14/19] Compiling SWXMLHash XMLAttributeDeserializable.swift
[15/19] Compiling Yams Constructor.swift
remark: Incremental compilation has been disabled: it is not compatible with whole module optimization
[16/20] Compiling ArgumentParser BashCompletionsGenerator.swift
[17/20] Compiling SourceKittenFramework ByteCount.swift
[18/20] Compiling SwiftSyntax AbsolutePosition.swift
remark: Incremental compilation has been disabled: it is not compatible with whole module optimization
remark: Incremental compilation has been disabled: it is not compatible with whole module optimization
remark: Incremental compilation has been disabled: it is not compatible with whole module optimization
remark: Incremental compilation has been disabled: it is not compatible with whole module optimization
[19/24] Compiling SwiftIDEUtils SwiftIDEUtilsCompatibility.swift
[20/24] Compiling SwiftBasicFormat BasicFormat.swift
[21/24] Compiling SwiftDiagnostics Convenience.swift
[22/24] Compiling SwiftParser Attributes.swift
remark: Incremental compilation has been disabled: it is not compatible with whole module optimization
remark: Incremental compilation has been disabled: it is not compatible with whole module optimization
[23/26] Compiling SwiftOperators Operator.swift
[24/26] Compiling SwiftParserDiagnostics DiagnosticExtensions.swift
remark: Incremental compilation has been disabled: it is not compatible with whole module optimization
[25/27] Compiling SwiftSyntaxBuilder ConvenienceInitializers.swift
remark: Incremental compilation has been disabled: it is not compatible with whole module optimization
[26/28] Compiling SwiftSyntaxMacros AbstractSourceLocation.swift
remark: Incremental compilation has been disabled: it is not compatible with whole module optimization
[27/29] Compiling SwiftSyntaxMacroExpansion BasicMacroExpansionContext.swift
remark: Incremental compilation has been disabled: it is not compatible with whole module optimization
[28/30] Compiling SwiftCompilerPluginMessageHandling CompilerPluginMessageHandler.swift
remark: Incremental compilation has been disabled: it is not compatible with whole module optimization
[29/31] Compiling SwiftCompilerPlugin CompilerPlugin.swift
remark: Incremental compilation has been disabled: it is not compatible with whole module optimization
[30/32] Compiling SwiftLintCoreMacros RuleConfigurationMacros.swift
[30/32] Linking SwiftLintCoreMacros
remark: Incremental compilation has been disabled: it is not compatible with whole module optimization
[32/33] Compiling SwiftLintCore RuleDocumentation.swift
rosetta error: Failed to open libRosettaRuntime
 <unknown>:0: warning: compiler plugin not loaded: '/private/var/folders/rj/r1dvm6cd12s1cwvfk70xzx780000gn/T/mint/github.com_realm_SwiftLint/.build/x86_64-apple-macosx/release/SwiftLintCoreMacros; failed to initialize
/private/var/folders/rj/r1dvm6cd12s1cwvfk70xzx780000gn/T/mint/github.com_realm_SwiftLint/Source/SwiftLintCore/Models/ViolationSeverity.swift:3:13: error: external macro implementation type 'SwiftLintCoreMacros.MakeAcceptableByConfigurationElement' could not be found for macro 'MakeAcceptableByConfigurationElement()'
public enum ViolationSeverity: String, Comparable, Codable, InlinableOptionType {
            ^
/private/var/folders/rj/r1dvm6cd12s1cwvfk70xzx780000gn/T/mint/github.com_realm_SwiftLint/Source/SwiftLintCore/Helpers/Macros.swift:18:14: note: 'MakeAcceptableByConfigurationElement()' declared here
public macro MakeAcceptableByConfigurationElement() = #externalMacro(
             ^
/private/var/folders/rj/r1dvm6cd12s1cwvfk70xzx780000gn/T/mint/github.com_realm_SwiftLint/Source/SwiftLintCore/Helpers/Macros.swift:7:14: warning: external macro implementation type 'SwiftLintCoreMacros.AutoApply' could not be found for macro 'AutoApply()'
public macro AutoApply() = #externalMacro(
             ^
/private/var/folders/rj/r1dvm6cd12s1cwvfk70xzx780000gn/T/mint/github.com_realm_SwiftLint/Source/SwiftLintCore/Helpers/Macros.swift:18:14: warning: external macro implementation type 'SwiftLintCoreMacros.MakeAcceptableByConfigurationElement' could not be found for macro 'MakeAcceptableByConfigurationElement()'
public macro MakeAcceptableByConfigurationElement() = #externalMacro(
             ^
/private/var/folders/rj/r1dvm6cd12s1cwvfk70xzx780000gn/T/mint/github.com_realm_SwiftLint/Source/SwiftLintCore/Helpers/Macros.swift:37:14: warning: external macro implementation type 'SwiftLintCoreMacros.SwiftSyntaxRule' could not be found for macro 'SwiftSyntaxRule(foldExpressions:explicitRewriter:)'
public macro SwiftSyntaxRule(foldExpressions: Bool = false, explicitRewriter: Bool = false) = #externalMacro(
             ^
/private/var/folders/rj/r1dvm6cd12s1cwvfk70xzx780000gn/T/mint/github.com_realm_SwiftLint/Source/SwiftLintCore/Models/SeverityConfiguration.swift:8:6: error: generic struct 'ConfigurationElement' requires that 'ViolationSeverity' conform to 'AcceptableByConfigurationElement'
    @ConfigurationElement(key: "severity")
     ^
/private/var/folders/rj/r1dvm6cd12s1cwvfk70xzx780000gn/T/mint/github.com_realm_SwiftLint/Source/SwiftLintCore/Models/RuleConfigurationDescription.swift:393:15: note: where 'T' = 'ViolationSeverity'
public struct ConfigurationElement<T: AcceptableByConfigurationElement & Equatable>: Equatable {
              ^
/private/var/folders/rj/r1dvm6cd12s1cwvfk70xzx780000gn/T/mint/github.com_realm_SwiftLint/Source/SwiftLintCore/Models/SeverityConfiguration.swift:8:6: note: did you mean to use '.rawValue'?
    @ConfigurationElement(key: "severity")
     ^
                         .rawValue
/private/var/folders/rj/r1dvm6cd12s1cwvfk70xzx780000gn/T/mint/github.com_realm_SwiftLint/Source/SwiftLintCore/Models/SeverityConfiguration.swift:8:6: error: generic struct 'ConfigurationElement' requires that 'ViolationSeverity' conform to 'AcceptableByConfigurationElement'
    @ConfigurationElement(key: "severity")
     ^
/private/var/folders/rj/r1dvm6cd12s1cwvfk70xzx780000gn/T/mint/github.com_realm_SwiftLint/Source/SwiftLintCore/Models/RuleConfigurationDescription.swift:393:15: note: where 'T' = 'ViolationSeverity'
public struct ConfigurationElement<T: AcceptableByConfigurationElement & Equatable>: Equatable {
              ^
/private/var/folders/rj/r1dvm6cd12s1cwvfk70xzx780000gn/T/mint/github.com_realm_SwiftLint/Source/SwiftLintCore/Models/SeverityConfiguration.swift:8:6: note: did you mean to use '.rawValue'?
    @ConfigurationElement(key: "severity")
     ^
                         .rawValue
/private/var/folders/rj/r1dvm6cd12s1cwvfk70xzx780000gn/T/mint/github.com_realm_SwiftLint/Source/SwiftLintCore/Models/SeverityConfiguration.swift:8:6: error: generic struct 'ConfigurationElement' requires that 'ViolationSeverity' conform to 'AcceptableByConfigurationElement'
    @ConfigurationElement(key: "severity")
     ^
/private/var/folders/rj/r1dvm6cd12s1cwvfk70xzx780000gn/T/mint/github.com_realm_SwiftLint/Source/SwiftLintCore/Models/RuleConfigurationDescription.swift:393:15: note: where 'T' = 'ViolationSeverity'
public struct ConfigurationElement<T: AcceptableByConfigurationElement & Equatable>: Equatable {
              ^
/private/var/folders/rj/r1dvm6cd12s1cwvfk70xzx780000gn/T/mint/github.com_realm_SwiftLint/Source/SwiftLintCore/Models/SeverityConfiguration.swift:8:6: note: did you mean to use '.rawValue'?
    @ConfigurationElement(key: "severity")
     ^
                         .rawValue
/private/var/folders/rj/r1dvm6cd12s1cwvfk70xzx780000gn/T/mint/github.com_realm_SwiftLint/Source/SwiftLintCore/Models/SeverityConfiguration.swift:2:15: error: type 'SeverityConfiguration<Parent>' does not conform to protocol 'Equatable'
public struct SeverityConfiguration<Parent: Rule>: SeverityBasedRuleConfiguration {
              ^
/private/var/folders/rj/r1dvm6cd12s1cwvfk70xzx780000gn/T/mint/github.com_realm_SwiftLint/Source/SwiftLintCore/Models/SeverityConfiguration.swift:9:9: note: stored property type '<<error type>>' does not conform to protocol 'Equatable', preventing synthesized conformance of 'SeverityConfiguration<Parent>' to 'Equatable'
    var severity = ViolationSeverity.warning
        ^
Swift.==:1:24: note: candidate would match if 'SeverityConfiguration<Parent>' conformed to 'RawRepresentable'
@inlinable public func == <T>(lhs: T, rhs: T) -> Bool where T : RawRepresentable, T.RawValue : Equatable
                       ^
Swift.FloatingPoint:2:24: note: candidate would match if 'SeverityConfiguration<Parent>' conformed to 'FloatingPoint'
    public static func == (lhs: Self, rhs: Self) -> Bool
                       ^
Swift.BinaryInteger:2:24: note: candidate would match if 'SeverityConfiguration<Parent>' conformed to 'BinaryInteger'
    public static func == <Other>(lhs: Self, rhs: Other) -> Bool where Other : BinaryInteger
                       ^
Swift._Pointer:2:24: note: candidate would match if 'SeverityConfiguration<Parent>' conformed to '_Pointer'
    public static func == (lhs: Self, rhs: Self) -> Bool
                       ^
Swift._Pointer:3:35: note: candidate would match if 'SeverityConfiguration<Parent>' conformed to '_Pointer'
    @inlinable public static func == <Other>(lhs: Self, rhs: Other) -> Bool where Other : _Pointer
                                  ^
Swift.Strideable:3:35: note: candidate would match if 'SeverityConfiguration<Parent>' conformed to 'Strideable'
    @inlinable public static func == (x: Self, y: Self) -> Bool
                                  ^
Swift.StringProtocol:2:35: note: candidate would match if 'SeverityConfiguration<Parent>' conformed to 'StringProtocol'
    @inlinable public static func == <RHS>(lhs: Self, rhs: RHS) -> Bool where RHS : StringProtocol
                                  ^
Swift.SIMD:4:24: note: candidate would match if 'SeverityConfiguration<Parent>' conformed to 'SIMD'
    public static func == (a: Self, b: Self) -> Bool
                       ^
Swift.Equatable:2:17: note: protocol requires function '==' with type '(SeverityConfiguration<Parent>, SeverityConfiguration<Parent>) -> Bool'
    static func == (lhs: Self, rhs: Self) -> Bool
                ^
/private/var/folders/rj/r1dvm6cd12s1cwvfk70xzx780000gn/T/mint/github.com_realm_SwiftLint/Source/SwiftLintCore/Models/SeverityConfiguration.swift:19:14: error: cannot assign to property: 'severity' is a get-only property
        self.severity = severity
        ~~~~~^~~~~~~~
/private/var/folders/rj/r1dvm6cd12s1cwvfk70xzx780000gn/T/mint/github.com_realm_SwiftLint/Source/SwiftLintCore/Models/SeverityConfiguration.swift:29:14: error: cannot assign to property: 'severity' is a get-only property
        self.severity = severity
        ~~~~~^~~~~~~~
/private/var/folders/rj/r1dvm6cd12s1cwvfk70xzx780000gn/T/mint/github.com_realm_SwiftLint/Source/SwiftLintCore/Models/ViolationSeverity.swift:3:13: error: type 'ViolationSeverity' does not conform to protocol 'AcceptableByConfigurationElement'
public enum ViolationSeverity: String, Comparable, Codable, InlinableOptionType {
            ^
/private/var/folders/rj/r1dvm6cd12s1cwvfk70xzx780000gn/T/mint/github.com_realm_SwiftLint/Source/SwiftLintCore/Models/RuleConfigurationDescription.swift:354:17: note: type 'ViolationSeverity' does not conform to inherited protocol 'AcceptableByConfigurationElement'
public protocol InlinableOptionType: AcceptableByConfigurationElement {}
                ^
error: fatalError
🌱 Encountered error during "swift build -c release --product swiftlint -Xswiftc -target -Xswiftc x86_64-apple-macosx13.5". Use --verbose to see full output
🌱  Failed to build SwiftLint 0.54.0 with SPM
iMac-mm-6201:newspicks-app-ios mu$ mint install realm/SwiftLint@0.54.0
🌱 Cloning SwiftLint 0.54.0
🌱 Resolving package
🌱 Building product swiftlint
```

### Environment

- SwiftLint version 0.54.0
- Installation method used Mint version 0.17.5.
- Paste your configuration file:

```yml
disabled_rules:
  - identifier_name
  - line_length
  - file_length
  - type_body_length
  - force_cast
  - type_name
  - cyclomatic_complexity
  - function_body_length
  - force_try
  - nesting
  - function_parameter_count
  - large_tuple

included:
  - xxx（Confidential information）
```

- Xcode Version 15.0

**Comment by jpsim** [comment]
> rosetta error: Failed to open libRosettaRuntime

Have you tried installing Rosetta?

**Comment by erikkerber** [comment]
@jpsim Very LQ: What is requiring Rosetta as of 0.54.0?

We don't install it on our CI, so wondering if there's a reason we should/need to going forward?

**Comment by SimplyDanny** [comment]
Nothing requires Rosetta in SwiftLint itself.

If I run `swift build -c release --product swiftlint -Xswiftc -target -Xswiftc x86_64-apple-macosx13.5` on an Apple Silicon Mac, I see the exact same error, even with Rosetta installed. The `x86_64-apple-macosx13.5` target seems wrong to me. If the Mint command is run on an Arm-based Mac, we wouldn't see the `x86_64`. On the other hand, running the command on an Intel-based Mac wouldn't require Rosetta.

@asa08: On which platform do you run the command?

**Comment by jpsim** [comment]
The main change that I think could cause this in 0.54.0 is the addition of Swift macros.

Does Mint have trouble installing other projects using macros?

**Comment by erikkerber** [comment]
Sorry I should have clarified. We aren't using Mint, we're using Bazel with bzlmod.

```starlark
bazel_dep(name = "swiftlint", version = "0.54.0")
```

We get the same Rosetta-ish error when (it appears) it's compiling macros

```
ERROR: /private/var/tmp/_bazel_buildkite/e21df498267afdbcbfd273d3ab407ad2/external/swiftlint~0.54.0/BUILD:48:14: Compiling Swift module @@swiftlint~0.54.0//:SwiftLintCore failed: (Exit 1): worker failed: error executing SwiftCompile command (from target @@swiftlint~0.54.0//:SwiftLintCore)
  (cd /private/var/tmp/_bazel_buildkite/e21df498267afdbcbfd273d3ab407ad2/execroot/_main && \
  exec env - \
    APPLE_SDK_PLATFORM=MacOSX \
    APPLE_SDK_VERSION_OVERRIDE=14.0 \
    PATH=/bin:/usr/bin:/usr/local/bin \
    SWIFT_AVOID_WARNING_USING_OLD_DRIVER=1 \
    XCODE_VERSION_OVERRIDE=15.0.0.15A240d \
    ZERO_AR_DATE=1 \
  bazel-out/darwin_x86_64-opt-exec-ST-13d3ddad9198/bin/external/rules_swift~1.13.0/tools/worker/worker swiftc @bazel-out/darwin_x86_64-dbg/bin/external/swiftlint~0.54.0/SwiftLintCore.swiftmodule-0.params)
# Configuration: d18ff59a88335a8e65c7576b51d3557be72a9bdefd3da972549ee68dd90e23ee
# Execution platform: @@local_config_platform//:host
error: emit-module command failed with exit code 1 (use -v to see invocation)
rosetta error: Failed to open libRosettaRuntime
 <unknown>:0: warning: compiler plugin not loaded: 
```

**Comment by jpsim** [comment]
Trying and failing to reproduce this.

```console
$ mint run realm/SwiftLint@0.54.0 version
0.54.0
```

So that works for me. I also tried with Bazel / bzlmod:

```console
$ git clone https://github.com/jpsim/swiftlint-bazel-example
$ cd swiftlint-bazel-example
$ bazel run -c opt @SwiftLint//:swiftlint -- version
INFO: Analyzed target @swiftlint~0.54.0//:swiftlint (0 packages loaded, 0 targets configured).
INFO: Found 1 target...
Target @swiftlint~0.54.0//:swiftlint up-to-date:
  bazel-bin/external/swiftlint~0.54.0/swiftlint
INFO: Elapsed time: 0.072s, Critical Path: 0.00s
INFO: 1 process: 1 internal.
INFO: Build completed successfully, 1 total action
INFO: Running command line: bazel-bin/external/swiftlint~0.54.0/swiftlint version
0.54.0
```

I'm on M1, macOS 14.1.1, with Xcode 15.0 (15A240d) installed and selected.

**Comment by SimplyDanny** [comment]
> I'm on M1, macOS 14.1.1, with Xcode 15.0 (15A240d) installed and selected.

Same for me. Both work. Cross-compiling with `swift build -c release --product swiftlint -Xswiftc -target -Xswiftc x86_64-apple-macosx13.5` produces the reported error though. Might be caused by SPM not yet supporting cross-compilation?

**Comment by pdrcabrod** [comment]
I was having the issue. Sharing my observations in case it helps somebody else

In my case the problem was that I had Homebrew installed for intel chipsets (prefix /usr/local). Migrating Homebrew to the M1 chipset, then install mint using the new Homebrew installation, did the trick

**Comment by erikkerber** [comment]
We were also spawning from an x86/Rosetta process (forgotten x86 Danger binary). Sorry for the added noise 🙇

**Comment by SimplyDanny** [comment]
Seems like this is resolved then. @asa08, feel free to reopen if not for you.

**Comment by yamataku29** [comment]
@SimplyDanny 
Hi Danny.
I have a similar problems is occurring in my environment. 

There're some fatalError in SeverityConfiguration.swift and ViolationSeverity.swift.
You can see it in attached images.

Could you tell me the solution?

![スクリーンショット 2024-02-06 16 01 23](https://github.com/realm/SwiftLint/assets/17525335/73121358-4451-4f1c-9472-547d9b1ce832)

![スクリーンショット 2024-02-06 16 01 26](https://github.com/realm/SwiftLint/assets/17525335/1afe553c-359c-413a-8b16-a2f94921757b)

I let you know some information text in bellow.

■fatalError
> /private/var/folders/51/lhm4698x3gzb7_lxrw5gcgvw0000gn/T/mint/github.com_realm_SwiftLint/Source/SwiftLintCore/Models/SeverityConfiguration.swift:2:15 type 'SeverityConfiguration<Parent>' does not conform to protocol 'Equatable'

> /private/var/folders/51/lhm4698x3gzb7_lxrw5gcgvw0000gn/T/mint/github.com_realm_SwiftLint/Source/SwiftLintCore/Models/SeverityConfiguration.swift:8:6 generic struct 'ConfigurationElement' requires that 'ViolationSeverity' conform to 'AcceptableByConfigurationElement'


> /private/var/folders/51/lhm4698x3gzb7_lxrw5gcgvw0000gn/T/mint/github.com_realm_SwiftLint/Source/SwiftLintCore/Models/SeverityConfiguration.swift:19:14 cannot assign to property: 'severity' is a get-only property

> /private/var/folders/51/lhm4698x3gzb7_lxrw5gcgvw0000gn/T/mint/github.com_realm_SwiftLint/Source/SwiftLintCore/Models/SeverityConfiguration.swift:29:14 cannot assign to property: 'severity' is a get-only property

> /private/var/folders/51/lhm4698x3gzb7_lxrw5gcgvw0000gn/T/mint/github.com_realm_SwiftLint/Source/SwiftLintCore/Models/ViolationSeverity.swift:3:13 external macro implementation type 'SwiftLintCoreMacros.MakeAcceptableByConfigurationElement' could not be found for macro 'MakeAcceptableByConfigurationElement()'

> /private/var/folders/51/lhm4698x3gzb7_lxrw5gcgvw0000gn/T/mint/github.com_realm_SwiftLint/Source/SwiftLintCore/Models/ViolationSeverity.swift:3:13 type 'ViolationSeverity' does not conform to protocol 'AcceptableByConfigurationElement'


```Mintfile
SwiftGen/SwiftGen@6.6.2
realm/SwiftLint@0.53.0
```

**Comment by yamataku29** [comment]
I send you more Information.
I did comment out like bellow.

```Build Phases Run Scriot
export PATH="/opt/homebrew/bin:$PATH"

if which mint >/dev/null; then
  #xcrun --sdk macosx mint run swiftlint swiftlint --fix
  #xcrun --sdk macosx mint run swiftlint swiftlint
  echo "testtest"
else
  echo "warning: Mint not installed, download from https://github.com/yonaskolb/Mint"
fi
```

Then, it start working.

**Comment by SimplyDanny** [comment]
0.54.0 is the first version requiring Swift 5.9 to make use of the macro feature.

@yamataku29: In your logs, we see that the macros cannot be resolved. I'm not a Mint user. So I cannot tell from my experience what's wrong.

**Comment by mactkg** [comment]
## tl;dr
If you installed mint with homebrew, reinstall homebrew without Rosetta.

## details
my co-worker was encountered this issue, but we could resolved with this approach.

https://github.com/realm/SwiftLint/issues/5348#issuecomment-1825389815

If homebrew was installed in Rosetta environment, homebrew is seems to install packages for x86_64 architecture even if Terminal launched without Rosetta.
In fact, `mint` binary is built for x86_64 arch.

```bash
$ file /usr/local/bin/mint
/usr/local/bin/mint: Mach-O 64-bit executable x86_64
```

Again, it can't change architecture to arm64 even if reinstall mint with non rosetta emulated Terminal.app. Reinstall homebrew for x86_64 arch(without rosetta) to resolve this issue.

Steps:
- Uninstall homebrew(in Rosetta environment)
- Install homebrew(in **non** Rosetta environment)
- Install mint by homebrew
- Install SwiftLint by mint
- It will work...!

**Comment by k06a** [comment]
Having similar issue:

```bash
mint run realm/SwiftLint@0.54.0 version
```
=>
```bash
🌱 Cloning SwiftLint 0.54.0
🌱 Resolving package
warning: could not determine XCTest paths: terminated(1): /usr/bin/xcrun --sdk macosx --show-sdk-platform-path output:
    xcrun: error: unable to lookup item 'PlatformPath' from command line tools installation
    xcrun: error: unable to lookup item 'PlatformPath' in SDK '/Library/Developer/CommandLineTools/SDKs/MacOSX.sdk'
    
error: 'github.com_realm_swiftlint': Invalid manifest (compiled with: ["/Library/Developer/CommandLineTools/usr/bin/swiftc", "-vfsoverlay", "/var/folders/xc/pq5jc85j6lx55bjmm1pzzcjm0000gn/T/TemporaryDirectory.leUMVO/vfs.yaml", "-L", "/Library/Developer/CommandLineTools/usr/lib/swift/pm/ManifestAPI", "-lPackageDescription", "-Xlinker", "-rpath", "-Xlinker", "/Library/Developer/CommandLineTools/usr/lib/swift/pm/ManifestAPI", "-target", "arm64-apple-macosx13.0", "-sdk", "/Library/Developer/CommandLineTools/SDKs/MacOSX.sdk", "-swift-version", "5", "-I", "/Library/Developer/CommandLineTools/usr/lib/swift/pm/ManifestAPI", "-sdk", "/Library/Developer/CommandLineTools/SDKs/MacOSX.sdk", "-package-description-version", "5.9.0", "/private/var/folders/xc/pq5jc85j6lx55bjmm1pzzcjm0000gn/T/mint/github.com_realm_SwiftLint/Package.swift", "-Xfrontend", "-disable-implicit-concurrency-module-import", "-Xfrontend", "-disable-implicit-string-processing-module-import", "-o", "/var/folders/xc/pq5jc85j6lx55bjmm1pzzcjm0000gn/T/TemporaryDirectory.9q94D0/github.com_realm_swiftlint-manifest"])
/private/var/folders/xc/pq5jc85j6lx55bjmm1pzzcjm0000gn/T/mint/github.com_realm_SwiftLint/Package.swift:2:8: error: no such module 'CompilerPluginSupport'
import CompilerPluginSupport
       ^
🌱 Encountered error during "swift package resolve". Use --verbose to see full output
🌱  Failed to resolve SwiftLint 0.54.0 with SPM
```

**Comment by fluiddruid** [comment]
I hit the same thing k06a. You've probably got the SDK path set to something other than the XCode instance. See here: [https://stackoverflow.com/a/43418980](https://stackoverflow.com/a/43418980)

Try running 

` sudo xcode-select -switch /Applications/Xcode.app/Contents/Developer`

Then rerun your mint command.
